docs/docs/developers/contracts/references/portals/outbox.md

```diff
@@ -6,20 +6,24 @@ The `Outbox` is a contract deployed on L1 that handles message passing from the
 
 **Links**: [Interface](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/interfaces/messagebridge/IOutbox.sol), [Implementation](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/messagebridge/Outbox.sol).
 
-## `sendL1Messages()`
+## `insert()`
 
-Inserts multiple messages from the `Rollup`.
+Inserts the root of a merkle tree containing all of the L2 to L1 messages in a block specified by _l2BlockNumber.
+
+#include_code outbox_insert l1-contracts/src/core/interfaces/messagebridge/IOutbox.sol solidity
 
-#include_code outbox_send_l1_msg l1-contracts/src/core/interfaces/messagebridge/IOutbox.sol solidity
 
 | Name           | Type    | Description |
 | -------------- | ------- | ----------- |
-| `_entryKeys`         | `bytes32[]` | A list of message hashes to insert into the outbox for later consumption |
+| `_l2BlockNumber` | `uint256` | The L2 Block Number in which the L2 to L1 messages reside |
+| `_root` | `bytes32` | The merkle root of the tree where all the L2 to L1 messages are leaves |
+| `_height` | `uint256` | The height of the merkle tree that the root corresponds to |
 
 #### Edge cases
 
-- Will revert with `Registry__RollupNotRegistered(address rollup)` if `msg.sender` is not registered as a rollup on the [`Registry`](./registry.md)
-- Will revert `Outbox__IncompatibleEntryArguments(bytes32 entryKey, uint64 storedFee, uint64 feePassed, uint32 storedVersion, uint32 versionPassed, uint32 storedDeadline, uint32 deadlinePassed)` if insertion is not possible due to invalid entry arguments.
+- Will revert with `Outbox__Unauthorized()` if `msg.sender != ROLLUP_CONTRACT`. 
+- Will revert with `Errors.Outbox__RootAlreadySetAtBlock(uint256 l2BlockNumber)` if the root for the specific block has already been set.
+- Will revert with `Errors.Outbox__InsertingInvalidRoot()` if the rollup is trying to insert a zeroed root.
```

Comment:
Would probably write `bytes32(0` as a root instead. Just to make clear that zeroed is not a root of tree with zeros.